pkg/proof/querier.go

```diff
@@ -34,6 +34,9 @@ func QueryTxInclusionProof(_ sdk.Context, path []string, req abci.RequestQuery)
 	if err != nil {
 		return nil, err
 	}
+	if index < 0 {
+		return nil, fmt.Errorf("path[0] element: %q produced a negative value: %d", path[0], index)
```

Comment:
[optional]I understand this is a backport, but for clarity in potential future PRs, it might be beneficial to specify that the "index" should be non-negative, either as an addition or an alternative to referencing the `path[0] element`.